Update social sharing features of documentation site and self-host fonts

justinbburris opened the pull request:
# Documentation Website Improvements

This PR includes several improvements to the documentation website, focusing on social sharing, performance, and code organization:

## Website Enhancements
- Added comprehensive meta tags for better social sharing
- Implemented Open Graph tags for improved social media sharing
- Added author metadata to pages
- Added world map sharing functionality

## JavaScript Refactoring
- Resolved SonarCloud warnings
- Reorganized JavaScript code into `fheroes2` namespace for better organization
- Standardized JavaScript manager class implementations
- Converted manager classes to use static methods to better represent their use
- Renamed files to follow consistent naming convention:
  - `accessibility.js` → `accessibility-manager.js`
  - `code-copy.js` → `code-copy-manager.js`
  - `heading-copy.js` → `heading-copy-manager.js`
  - `theme-toggle.js` → `theme-manager.js`

## Font Management
- Switched to self-hosted fonts for better maintainability
- Added local font files for:
  - Asul (Regular and Bold)
  - Lancelot
  - Jacquard 24
- Included proper font licensing (OFL) for all fonts

closes https://github.com/ihhub/fheroes2/pull/9736#issuecomment-2827066646
closes https://github.com/ihhub/fheroes2/pull/9736#discussion_r2057628339

## Screenshots

### Social Sharing example
<img width="722" alt="Screenshot_2025-04-26_at_3_02_21 PM" src="https://github.com/user-attachments/assets/478639fb-0bc1-4f0c-87a4-6527012009ff" />

justinbburris commented:
cc @oleg-derevenetz 
I hooked up sonarcloud for my repo and got a clean run, after I fixed the errors raised by the tooling.
https://github.com/justinbburris/fheroes2/actions/runs/14676449087/job/41193350553

oleg-derevenetz commented on docs/README.md:
```diff
@@ -11,7 +11,7 @@ You can find a complete list of all of our changes and enhancements in [**its ow
 <div class="image-showcase">
     <div class="row">
-        <img src="images/screenshots/screenshot_world_map.png?raw=true" width="820px" class="full-image game-screenshot" alt="Screenshot of the world map">
+        <img src="images/screenshots/screenshot_world_map.png?raw=true" width="820" class="full-image game-screenshot" alt="Screenshot of the world map">
```
Not related to your recent changes, just a long-standing bug :) The `width` attribute of the `<img>` element [should always be in pixels without a unit](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/img#width).

justinbburris replied:
well good to know! Thanks for pointing it out!

oleg-derevenetz commented on docs/assets/js/code-copy.js:
```diff
@@ -103,6 +103,9 @@ class CodeCopyManager
+        return;
+    }
+
+    window.codeCopyManager = new CodeCopyManager();
```
Not a call to action, just a question :) Are there any "best practices" for naming custom window properties (maybe starting with an underscore or something like that) to avoid name collisions with standard properties that may be added in the future?

justinbburris replied:
super great callout - 2 things

1. We don't _really_ need these classes to be instantiated - it's mostly to encapsulate the logic from the event. So we can make them static and avoid the window issue. This wont work for the `AccessibilityManager`, as it is used by these classes, so we'll need it in the window.
2. That said, with these window objects we really should be encapsulating them in a namespace, to prevent the issue you're raising from happening. So typically we'd put it into a project namespace, like `fheroes2` - which will be a WILD day if that is ever a standard property. Great call - I'll update the PR with that

oleg-derevenetz commented on sonar-project.properties:
```diff
@@ -21,9 +21,14 @@ sonar.sources=.
+# Ignore Web:PageWithoutTitleCheck ("<title>" should be present in all pages) for
+# the default.html template (thig element will be added later by Jekyll).
+sonar.issue.ignore.multicriteria.webPageWithoutTitleCheckForDefaultTemplate.ruleKey=Web:PageWithoutTitleCheck
+sonar.issue.ignore.multicriteria.webPageWithoutTitleCheckForDefaultTemplate.resourceKey=docs/_layouts/default.html
```
That "<title> should be present in all pages" Sonar bug was a false positive, since that element is added later by Jekyll, so I removed the redundant element and ignored this "bug".

justinbburris replied:
You know. I was thinking the same thing. Thanks for making the call.

oleg-derevenetz reviewed (approved):
Hi @justinbburris I appreciate your careful approach :) I just made a few changes and asked one question. Could you please take a look in your spare time to check these changes?

P.S. I wouldn't call it regression, because nothing that worked before was broken :)

justinbburris commented:
okay @oleg-derevenetz I've implemented the changes to the objects on the window and uh.. a few other things.

Here's the latest [sonar run](https://github.com/justinbburris/fheroes2/actions/runs/14684220698/job/41210767272?pr=6)

FYI - I've done what might be a little bit of faux pas, in that I rebased off `master` and force pushed (to remove my sonarcloud changes without a revert commit) I figured that there probably wouldn't be too many more changes after the fact, but I wanted to make sure you were aware. The history still looks 👍 but LMK if you'd rather I not do that going forward.

Also, I've folded in some feedback from @ihhub in the previous PR (moving the fonts) and improved the social sharing a bit. That is all detailed in the PR description.

oleg-derevenetz reviewed (approved):
Hi @justinbburris looks good, thank you!

> FYI - I've done what might be a little bit of faux pas, in that I rebased off master and force pushed (to remove my sonarcloud changes without a revert commit) I figured that there probably wouldn't be too many more changes after the fact, but I wanted to make sure you were aware. The history still looks 👍 but LMK if you'd rather I not do that going forward.

You are free to remove, reorder or merge multiple commits into one in the PR context if you think this will better showcase the logic of a development process.

ihhub commented on docs/_layouts/default.html:
```diff
@@ -5,13 +5,38 @@
+    />
+    <meta
+      property="og:image"
+      content="{{ '/images/screenshots/screenshot_world_map.png' | absolute_url }}"
```
Is it correct that this is a link to an image which will be displayed in a URL preview and it seem that any browser / app would attempt to download the whole image? If it is true do you think that we need to make a smaller version of the image to reduce the download time for link previews?

oleg-derevenetz replied:
Hi @ihhub the "Best Practices" page for the Facebook Sharing (these are the ones who invented all these tags) [recommends](https://developers.facebook.com/docs/sharing/best-practices) the preview image to be at least 1080px in width for best display. The absolute minimum is 600px. Our screenshot is currently 1024px. I slightly reduced the screenshot sizes in bytes by my latest commit here (about 5-10%).

justinbburris replied:
Thanks so much @oleg-derevenetz you beat me to it :)

justinbburris replied:
There are actually a few additional best practices in there that we can implement too

ihhub replied:
Well done, @oleg-derevenetz !

ihhub commented on docs/assets/fonts/Jacquard_24/Jacquard24-Regular.ttf:
This font is 300KB in size. In comparison it is way bigger than others (30-45 KB). It seems that this font is used only for 1 thing: `fheroes2` title (h1). Do you think that we can find a way to optimize the size of this font to help rendering `fheroes2` character string? I know that this might seen like an overkill in optimization but sometimes it is important to deliver the best performance on all levels :)

justinbburris replied:
I'll do some experimentation tonight and see if I can get a good result here.

oleg-derevenetz replied:
Hi @ihhub I suppose this issue can be addressed separately later, if someone can find a suitable font file of a smaller size?

ihhub replied:
Yes, we can address it later.

ihhub replied:
@justinbburris , please do. No harm to make another PR :)

justinbburris replied:
Done! 💪 See #9777 :)

ihhub reviewed (changes requested):
Hi @justinbburris , I put 2 comments in this pull request. Could you please take a look at them in your free time?

ihhub commented:
@justinbburris , thank you very much for this update!
